Fixes #16

tgross-usgs opened the pull request:
Added new Network endpoint on stations controller to return NLDI stations located with search distance on the network. I created two service agents for each of the services (Navigation Services and NLDI Services).  I've been testing with the following requests, modifying the search distances to test all scenarios (no upstream sites, no downstream sites, no sites within distance, sites both upstream and downstream). 

http://localhost:53812/stations/Network?lat=41.606715&lon=-103.010559&distance=500
http://localhost:53812/stations/Network?lat=38.263928&lon=-104.551306&distance=200

I also took a guess and set this up to hit NLDI with a search for stations **downstream with diversions** and **upstream including tributaries**. If these are incorrect they can easily be changed. The options we have for NLDI navigate searches are below.

- DD is __D__ownstream navigation with __D__iversions
- DM is __D__ownstream navigation on the __M__ain channel
- UM is __U__pstream navigation on the __M__ain channel
- UT is __U__pstream navigation including all __T__ributaries

Commit:
Added new Network endpoint on stations controller to return NLDI stations located with search distance on the network

Commit:
merged responses from NLDI

Commit:
add documentation for nearest gage and nearest gage on a network endpoints and removed pagination for the nearest gage endpoint

kjacobsen16 reviewed (approved):
@tgross-usgs just a couple small comments:
1. There is one hard-coded url in the NLDIServiceAgent. Can we pull that out into the appsettings.json? Generally any service calls we make are best put in there to make it easier to pinpoint/for us to check what services are implemented :) 
2. We should put a description of the "distance" parameter in the appsettings.json. It also seems like it's km, whereas the nearest by radius is m. We should maybe pick one and go with it? 

Otherwise working well! We may want more information from the DB, but for now having the station number enables us to query the services for that station. We'll also probably have to talk more about if we need consistent response formats, etc. at some point.

Commit:
deleted unused method from NLDIServiceAgent, updated Nearest endpoint to use kilometers for search radius

tgross-usgs commented:
@kjacobsen16 below are my responses...
1. This was actually an unused method in the NLDIServiceAgent. I had it set up for my original testing and forgot to delete during cleanup. The url components already exist in appsettings and are populated in the NavigationAgent so I simply deleted the entire method. Sorry I missed it during my first attempt at clean-up.
2. The units were different because the default used by the GageStatsDB is meters but the NLDI services default is km. I updated the Nearest endpoint so that it is now expecting km from the user. For consistency both endpoints will request kilometers. Good catch. 
3. I've updated the parameters and units in appsettings and the units in the documentation markdown.
